### PR TITLE
NO-TICKET: Faster trie integrity checks

### PR DESCRIPTION
**Pull request by xcthulhu**

### CODE REVIEW

File: execution_engine/src/storage/trie_store/operations/mod.rs [hunk 1]
```diff
@@ -241,7 +246,11 @@ where
     E: From<S::Error> + From<bytesrepr::Error>,
 {
     let mut missing_descendants = Vec::new();
+    let mut visited = HashSet::new();
     while let Some(trie_key) = trie_keys_to_visit.pop() {
+        if !visited.insert(trie_key.clone()) {
```
Comment (afck) [inline execution_engine/src/storage/trie_store/operations/mod.rs:251]: No need to clone, says Clippy.

**Review by afck — APPROVED** [review]

**Comment by goral09** [comment]
We can also only read the last `UNBONDING_DELAY * 2 + 1` switch blocks and verify the integrity of their state root hashes.

File: execution_engine/src/storage/trie_store/operations/mod.rs (hunk 1 above)
Comment (xcthulhu) [reply execution_engine/src/storage/trie_store/operations/mod.rs]: Fixed in https://github.com/CasperLabs/casper-node/pull/1074/commits/99742d03286e7e825ef94537ef644bd016db9ec6

**Comment by xcthulhu** [comment]
> We can also only read the last UNBONDING_DELAY * 2 + 1 switch blocks and verify the integrity of their state root hashes.

Might break things when people read under a particular state root in the RPC...

**Pull request merged**